pallets/ddc-verification/src/lib.rs

```diff
@@ -636,6 +666,98 @@ pub mod pallet {
 		pub(crate) number_of_gets: u64,
 	}
 
+	/// Challenge Response
+	#[derive(
+		Debug, Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Encode, Decode,
+	)]
+	pub(crate) struct ChallengeAggregateResponse {
+		/// proofs
+		pub proofs: Vec<Proof>,
+	}
+
+	#[derive(
+		Debug, Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Encode, Decode,
+	)]
+	pub(crate) struct Proof {
+		pub merkle_tree_node_id: u64,
+		pub usage: Usage,
+		pub path: Vec<String>,
+		pub leafs: Vec<Leaf>,
+	}
+
+	#[derive(
+		Debug, Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Encode, Decode,
+	)]
+	pub(crate) struct Usage {
+		/// Total amount of stored bytes.
+		pub stored_bytes: i64,
+		/// Total amount of transferred bytes.
+		pub transferred_bytes: u64,
+		/// Total number of puts.
+		pub number_of_puts: u64,
+		/// Total number of gets.
+		pub number_of_gets: u64,
+	}
+
+	#[derive(
+		Debug, Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Encode, Decode,
+	)]
+	pub(crate) struct Leaf {
+		pub record: Record,
+		pub transferred_bytes: u64,
+		pub stored_bytes: i64,
+	}
+
+	#[derive(
+		Debug, Serialize, Deserialize, Clone, Hash, Ord, PartialOrd, PartialEq, Eq, Encode, Decode,
+	)]
+	#[allow(non_snake_case)]
+	pub(crate) struct Record {
```

Comment:
Would be better to replicate protobuf message version in a next version.

https://github.com/Cerebellum-Network/ddc-storage-node/blob/dev/pb/activity_record.proto

Reply:
I mean the next version of validation can define Record type similar to protobuf message. Also the access requested.